src/bot.rs

```diff
@@ -322,27 +348,31 @@ impl RcBot {
             })
             .filter_map(move |mut val| {
                 debug!("Got an update from Telegram: {:?}", val);
-                let mut forward: Option<String> = None;
+
+                let mut sndr: Option<UnboundedSender<(RcBot, objects::Message)>> = None;
 
                 if let Some(ref mut message) = val.message {
                     if let Some(text) = message.text.clone() {
                         let mut content = text.split_whitespace();
                         if let Some(cmd) = content.next() {
-                            if self.inner.handlers.borrow_mut().contains_key(cmd) {
+                            if let Some(sender) = self.inner.handlers.borrow_mut().get_mut(cmd) {
+                                sndr = Some(sender.clone());
                                 message.text = Some(content.collect::<Vec<&str>>().join(" "));
-
-                                forward = Some(cmd.into());
+                            } else if let Some(ref mut sender) =
```

Comment:
Perhaps we should check if the command really begins with an "/", otherwise any input will be interpreted as a command.

Reply:
Is that bad? If the user have registered command "foo", I guessd the expected behaviour is to call handler when the exact match recieved.

Alternatively we can check that command in `new_cmd` function and prepend slash if there is no one.

Reply:
According to the Telegram docs a command should start with a slash (https://core.telegram.org/bots#commands). I think the best way forward is to check the first word for a slash and only then pass it as a command. And, as you said, enforce that the argument in new_cmd actually starts with a slash. But this can be done in a sperate commit.